ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -946,3 +891,393 @@ def test_ipa_hbac_rule_duplication(self):
+
+    def test_check_vault_is_migrated(self):
+        """
+        This testcase checks that vault is
+        not migrated
+        """
+        vault_name = "testvault"
+        CMD_OUTPUT = (
+            "----------------\n"
+            "0 vaults matched"
+            "----------------\n"
+            "----------------------------\n"
+            "Number of entries returned 0\n"
+            "----------------------------"
+        )
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
+            ["ipa", "vault-find", vault_name], raiseonerr=False)
+        assert cmd.returncode != 0
+        assert CMD_OUTPUT not in cmd.stdout_text
+
+
+class TestIPAMigrationStageMode(MigrationTest):
+    """
+    Tests for ipa-migrate tool in stage mode
+    """
+    def test_ipa_migrate_stage_mode(self, empty_log_file):
+        """
+        This test checks that ipa-migrate is successful
+        in dry run mode
+        """
+        tasks.kinit_admin(self.master)
+        tasks.kinit_admin(self.replicas[0])
+        MIGRATION_SCHEMA_LOG_MSG = "Migrating schema ...\n"
+        MIGRATION_CONFIG_LOG_MSG = "Migrating configuration ...\n"
+        IPA_UPGRADE_LOG_MSG = (
+            "Running ipa-server-upgrade ... (this may take a while)\n"
+        )
+        SIDGEN_TASK_LOG_MSG = "Running SIDGEN task ...\n"
+        MIGRATION_COMPLETE_LOG_MSG = "Migration complete!\n"
+        result = run_migrate(
+            self.replicas[0],
+            "stage-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n'],
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert result.returncode == 0
+        assert MIGRATION_SCHEMA_LOG_MSG in install_msg
+        assert MIGRATION_CONFIG_LOG_MSG in install_msg
+        assert IPA_UPGRADE_LOG_MSG in install_msg
+        assert SIDGEN_TASK_LOG_MSG in install_msg
+        assert MIGRATION_COMPLETE_LOG_MSG in install_msg
+
+    def test_ipa_migrate_stage_mode_new_user(self):
+        """
+        This testcase checks that when a new user is added and
+        ipa-migrate is run in stage-mode, uid/gid of the
+        migrated user is not preserved i.e we have different
+        uid/gid for user on remote and local IPA server.
+        """
+        username = 'stageuser1'
+        original_passwd = 'Secret123'
+        base_dn = str(self.master.domain.basedn)
+        LOG_MSG1 = (
+            "DEBUG Resetting the DNA range for new entry: "
+            "uid={},cn=users,cn=accounts,{}\n"
+        ).format(username, base_dn)
+        tasks.user_add(self.master, username, password=original_passwd)
```

Comment:
The user must be added in the prepare_ipa_server method as the ipa-migrate command is called only once, and before this test. You don’t really need a new user, you can use one of the testuser%d already created.